keyboards/peej/lumberjack/lumberjack.c

```diff
@@ -16,6 +16,37 @@
 
 #include "lumberjack.h"
 
+typedef struct {
+    int pin;
+    bool state;
+} led_state_t;
+
+int led_pattern_position = 0;
+led_state_t led_pattern[] = {
+    { LED1, true },
+    { LED2, true },
+    { LED1, false },
+    { LED2, false },
+    { LED1, true },
+    { LED2, true },
+    { LED1, false },
+    { LED2, false }
+};
+
+static uint32_t led_dance(uint32_t trigger_time, void* cb_arg) {
+    led_state_t led_state = led_pattern[led_pattern_position];
+    writePin(led_state.pin, led_state.state);
+    led_pattern_position++;
+    return led_pattern_position < sizeof(led_pattern) ? 100 : 0;
```

Comment:
Does this `sizeof`work correctly? Im a C noob so i dont know if the array is actually storing the structs or pointers to them, i would go with `ARRAY_SIZE` and add parenthesis just to safer
```suggestion
    return (led_pattern_position < ARRAY_SIZE(led_pattern)) ? 100 : 0;
```

Reply:
`ARRAY_SIZE(arg)` macro != `sizeof(arg)` function

Reply:
I know, and that's exactly why i said it... 
```c
// run at an online compiler
printf("%ld", sizeof(led_pattern)); // 64
printf("%ld", ARRAY_SIZE(led_pattern)); // 8
```

So... this code would loop over the array 8 times, but code makes it seem like you only loop once. 

?> Also, shouldn't `int` be `pin_t` even if it is `typedef`'ed to the same thing? (im not sure what it is internally)

---

`struct` + array epresentation doesn't add value because we are only toggling the LEDS, nds up being ode duplication... Wouldn't this be better?
```c
static uint32_t led_dance(uint32_t trigger_time, void* cb_arg) {
    static bool    state   = false;
    static uint8_t counter = 64; // whatever you want
    
    // change leds
    writePin(LED1, state);
    wirtePin(LED2, state);
    
    // update values for next iteration
    state = !state;
    --counter;
    
    // check if finished
    return counter ? 100 : 0;
}
```

Reply:
Agreed, doing something like this does simplify the code if it does reduce the clarity of what the LED sequence is.

On that note and after some more thought, this does all feel like it has gotten a lot more complex than my original goal of indicating the MCU has power and is flashed. So I've removed the deferred function calls and now just set the LEDs on in pre_init and off again in post_init. I think this accomplished the same goal, if with a little less razamataz.